src/SignalR/specs/HubProtocol.md

```diff
@@ -133,6 +133,10 @@ Ping messages do not have any payload, they are completely empty messages (aside
 
 The default ASP.NET Core implementation automatically pings both directions on active connections. These pings are at regular intervals, and allow detection of unexpected disconnects (for example, unplugging a server). If the client detects that the server has stopped pinging, the client will close the connection, and vice versa. If there's other traffic through the connection, keep-alive pings aren't needed. A `Ping` is only sent if the interval has elapsed without a message being sent.
 
+## Upload streaming
+
+Sometimes Callees will want to receive streaming data from Callers. The Caller can begin such a process by making an `Invocation` or `StreamInvocation` and adding a "StreamIds" property with an array of IDs. The IDs must be unique from any other stream IDs used by the same Caller. The Caller then sends `StreamItem` messages with the "InvocationId" property set to the ID for the stream they are sending over. The Caller ends the stream by sending a `Completion` message with the ID of the stream they are completing.
```

Comment:
The Callees doesn't have to initiate this though, right? I think this should be worded more like, The Caller has the ability to to send streaming data to the callee as well , or something.

Reply:
nit: for the line reading "and adding a "StreamIds" property with an array of IDs." Add something to the effect of  -> representing the 1 or more streams that will be associated with this invocation

Reply:
>The Callees doesn't have to initiate this though, right?

Initiate what?

Reply:
When you write "Sometimes Callees will want to receive streaming data from Callers"